cmd/kubeadm/app/phases/upgrade/postupgrade_test.go

```diff
@@ -169,6 +174,110 @@ func TestWriteKubeletConfigFiles(t *testing.T) {
 	}
 }
 
+func TestUnupgradedControlPlaneInstances(t *testing.T) {
+	testCases := []struct {
+		name          string
+		pods          []corev1.Pod
+		currentNode   string
+		expectedNodes []string
+		expectError   bool
+	}{
+		{
+			name: "one unupgraded instance",
+			pods: []corev1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kube-apiserver-1",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"component": constants.KubeAPIServer,
+						},
+					},
+					Spec: corev1.PodSpec{
+						NodeName: "node-1",
+						Containers: []corev1.Container{
+							{Name: constants.KubeAPIServer, Image: "k8s.gcr.io/kube-apiserver:v1"},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "kube-apiserver-2",
+						Namespace: metav1.NamespaceSystem,
+						Labels: map[string]string{
+							"component": constants.KubeAPIServer,
+						},
+					},
+					Spec: corev1.PodSpec{
+						NodeName: "node-2",
+						Containers: []corev1.Container{
+							{Name: constants.KubeAPIServer, Image: "k8s.gcr.io/kube-apiserver:v2"},
+						},
+					},
+				},
+			},
+			currentNode:   "node-1",
+			expectedNodes: []string{"node-2"},
+			expectError:   false,
+		},
+		{
+			name: "all upgraded instances",
+			pods: []corev1.Pod{
```

Comment:
in my humble opinion, maybe returning two pods with the same v1 image is better? 
1. because it's more clear to show the difference between the previous case. it kinda like control variable method.
2. additional, one node / mulit-node are totoally difference scenarios for kubeadm. maybe we should not mix them up.